covid19_npis/model/number_of_tests.py

```diff
@@ -15,10 +16,85 @@
     MvStudentT,
     MvNormalCholesky,
     Deterministic,
+    VonMises,
 )
 from .. import transformations
 from . import utils
 
+def weekly_modulation(name,modelParams,cases,week_modulation_type="abs_sine"):
+    r"""
+    Adds a weekly modulation of the number of new cases:
+    .. math::
+        \text{new\_cases} &= \text{new\_cases\_raw} \cdot (1-f(t))\,, \qquad\text{with}\\
+        f(t) &= f_w \cdot \left(1 - \left|\sin\left(\frac{\pi}{7} t- \frac{1}{2}\Phi_w\right)\right| \right),
+    if ``week_modulation_type`` is ``"abs_sine"`` (the default). If ``week_modulation_type`` is ``"step"``, the
+    new cases are simply multiplied by the weekend factor on the days set by ``weekend_days``
+    The weekend factor :math:`f_w` follows a Lognormal distribution with
+    median ``pr_mean_weekend_factor`` and sigma ``pr_sigma_weekend_factor``. It is hierarchically constructed if
+    the input is two-dimensional by the function :func:`hierarchical_normal` with default arguments.
+    The offset from Sunday :math:`\Phi_w` follows a flat :class:`~pymc3.distributions.continuous.VonMises` distribution
+    and is the same for all regions.
+    Parameters
+    ----------
+    cases : :class:`~theano.tensor.TensorVariable`
+        The input array of daily new cases, can be one- or two-dimensional
+    name_cases : str or None,
+        The name under which to save the cases as a trace variable.
+        Default: None, cases are not stored in the trace.
+    week_modulation_type : str
+        The type of modulation, accepts ``"step"`` or  ``"abs_sine`` (the default).
+    pr_mean_weekend_factor : float
+        Sets the prior mean of the factor :math:`f_w` by which weekends are counted.
+    pr_sigma_weekend_factor : float
+        Sets the prior sigma of the factor :math:`f_w` by which weekends are counted.
+    weekend_days : tuple of ints
+        The days counted as weekend if ``week_modulation_type`` is ``"step"``
+    model : :class:`Cov19Model`
+        if none, it is retrieved from the context
+    Returns
+    -------
+    new_cases : :class:`~theano.tensor.TensorVariable`
+    """
+
+    log.debug("Week modulation")
+
+    """
+    # TODO: - general check of function
+            - which shape does the modulation have to be? i guess time,country,age?
+            - check prior parameters
+            - check: which ones should be modulated? (mostly: positive tests!)
+            - different modulations across: age, country?
+            - check: are (cumulative) case numbers same as in unmodulated case? need some kind of normalization?
+            - store and plot parameters at end
```

Comment:
You should be able to add the name of the parameter to the plotting list (dummy_model.py bottom).